app/javascript/components/Navigation/MatchLogger/HeadToHeadLogger.jsx

```diff
@@ -0,0 +1,49 @@
+import React, { useEffect, useState } from 'react';
+import Button from '../../Button';
+import PlayerSelect from '../PlayerSelect';
+
+const MatchLogger = ({ unlockedUsers, currentUserId }) => {
+  const currentUserObject = unlockedUsers.find((user) => user.id === currentUserId);
+  const sortedUsers = unlockedUsers.sort((a, b) => a.name.localeCompare(b.name));
+
+  const [playerA, setPlayerA] = useState(currentUserObject);
+  const [playerB, setPlayerB] = useState(currentUserObject);
+  const [winnerId, setWinnerId] = useState(currentUserObject.id);
+  const [selectedDate, setSelectedDate] = useState(new Date().toLocaleDateString('en-CA'));
+  const [selectedTime, setSelectedTime] = useState(new Date().toLocaleTimeString('en-US', { hour12: false, hour: '2-digit', minute: '2-digit' }));
+
+  useEffect(() => {
+    if (winnerId !== playerA.id && winnerId !== playerB.id) {
+      setWinnerId(playerA.id);
+    }
+  }, [playerA, playerB]);
+
+  return (
+    <>
+      <p>Who played?</p>
+      <PlayerSelect
+        player={playerA}
+        players={sortedUsers}
+        setSelectedPlayer={setPlayerA}
+      />
+      <PlayerSelect
+        player={playerB}
+        players={sortedUsers}
+        setSelectedPlayer={setPlayerB}
+      />
+      <p>Who won?</p>
+      <select className="browser-default winner-selector" disabled={playerA === playerB} name="match[1]" onChange={(e) => setWinnerId(Number.parseInt(e.target.value, 10))}>
+        <option value={playerA.id}>{playerA.name}</option>
+        <option value={playerB.id}>{playerB.name}</option>
+      </select>
+      <input type="hidden" name="match[2]" value={winnerId === playerA.id ? playerB.id : playerA.id} />
```

Comment:
These "name" fields not correspond to the new way of logging matches. `match[1]` will always be the winner id. `match[2]` will always be the loser. We use the hidden field here in order to modify the form submission without necessarily needing user input (since we can always infer the loser from the winner selection)